src/display/annotation_layer.js

```diff
@@ -1880,6 +1885,10 @@ class PopupAnnotationElement extends AnnotationElement {
       this.elements.map(e => e.data.id).join(",")
     );
 
+    for (const element of this.elements) {
+      element.popup = popup;
+    }
```

Comment:
We're now looping through `this.elements` twice in a row, which seems slightly unnecessary; how about the following instead?
```suggestion

    const ids = [];
    for (const element of this.elements) {
      ids.push(element.data.id);
      element.popup = popup;
    }
    this.container.setAttribute("aria-controls", ids.join(","));
```